src/Fail.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace PhpWorkshop\PhpWorkshop;
+
+use PhpWorkshop\PhpWorkshop\Exercise\ExerciseInterface;
+
+/**
+ * Class Fail
+ * @package PhpWorkshop\PhpWorkshop
+ * @author Aydin Hassan <aydin@hotmail.co.uk>
+ */
+class Fail
```

Comment:
Not sure about these `Fail` and `Success` class. They don't inherit from any base so I can't type hint in `ResultAggregator`. If they were to extend or implement something, what would it be? I don't like the idea of combining them because `Success` doesn't require a `$reason`.